test/integration/multiple-done.spec.js

```diff
@@ -119,20 +119,18 @@ describe('multiple calls to done()', function() {
     });
 
     it('correctly attributes the errors', function() {
-      expect(res.failures, 'to satisfy', [
-        {
-          fullTitle: 'suite "before each" hook in "suite"',
-          err: {
-            message: /done\(\) called multiple times in hook <suite "before each" hook> of file.+multiple-done-before-each\.fixture\.js/
-          }
-        },
-        {
-          fullTitle: 'suite "before each" hook in "suite"',
-          err: {
-            message: /done\(\) called multiple times in hook <suite "before each" hook> of file.+multiple-done-before-each\.fixture\.js/
-          }
+      expect(res.failures[0], 'to equal', res.failures[1]);
+      expect(res.failures[0], 'to satisfy', {
+        fullTitle: 'suite "before each" hook in "suite"',
+        err: {
+          message: /done\(\) called multiple times in hook <suite "before each" hook in "suite"> of file.+multiple-done-before-each\.fixture\.js/,
+          multiple: [
```

Comment:
this error object is weird, but I guess that's what it does.

you can combine this into a single assertion:

```js
expect(res.failures[0], 'to equal', res.failures[1])
  .and('to satisfy', {
    // etc.
  });
```

Reply:
(likewise the assertion(s) in the new test may be able to be combined this way)

Reply:
Yeah, you probably already know this, but I rewrote the assertion to point out that both items of the array actually refer to the same object.  I can't do a `===` check in the tests, because it's been converted to/from JSON, so the assertions are running against 2 different parsed copies of the same object.  Does `'to equal'` handle that?

This happens because when a single test/hook has multiple failures, the first error is reused, and the `multiple` array is created to store the others.  Ideally I think mocha should be creating a new "hookinvocation" instance for each time the hook runs, where each "hookinvocation" can store a reference to the test case it's paired to.  But I don't think I'll have time for that size of code change.